File: Java/Inheritance/InheritCode.java
//Program to study inheritance of code

class A
{
 //methods
 void fx1()
 {
  System.out.println("A fx1()");
 }
 
 void fx2()
 {
  System.out.println("A fx2()");
 }
}//A

class B extends A	
{
  //redefine and inherited non static method (maintain signature)
  //i.e. override
  void fx2()
  {
   System.out.println("B fx2()");
  }

  void fx3()
  {
   System.out.println("B fx3()");
  }

}//B


class InheritCode
{
 public static void main(String args[])
 {
   //object of derived class

   B objB = new B();
   objB.fx1();//inherited code
   objB.fx2();//overridden code
   objB.fx3();//extended code
  
 }
}//InheritCode

File: c++/Inheritance/InheritCode.cpp
//Demonstration of inheritance of code.

#include<iostream>
using namespace std;


class A
{
public:
  void f1();
  void f2();
};

void A :: f1()
{
  cout<<"\n A f1";
}

void A :: f2()
{
  cout<<"\n A f2";
}

//derived entity B inherits A
class B : public A
{
 public:
   void f2();//overridden code
   void f3();//extended code
};


void B :: f2()
{
  cout<<"\n B f2 ";
}


void B :: f3()
{
  cout<<"\n B f3 ";
}


int main()
{
  B objB;
  objB.f1();//inherited code
  objB.f2();//overridden code
  objB.f3();//extended code

  return 0;
}
